src/MarkdownWriter/CommandHelpMarkdownWriter.cs

```diff
@@ -0,0 +1,242 @@
+    internal class CommandHelpMarkdownWriter
+    {
+        private readonly string _filePath;
+        private StringBuilder sb;
+        private readonly Encoding _encoding;
+
+        /// <summary>
+        /// Initialize the writer with settings.
+        /// </summary>
+        /// <param name="settings">Settings needs for the markdown writer.</param>
+        public CommandHelpMarkdownWriter(MarkdownWriterSettings settings)
+        {
+            string path = settings.DestinationPath;
+
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+            else
+            {
+                _filePath = path;
+                _encoding = settings.Encoding;
+                sb = Constants.StringBuilderPool.Get();
+            }
+        }
+
+        /// <summary>
+        /// Write the CommandHelp object to the specified path in the the settings during initialization.
+        /// The file is overwritten and not appended as we do not expect more than one CommandHelp per file.
+        /// </summary>
+        /// <param name="help">CommandHelp object to write.</param>
+        /// <param name="noMetadata">Specify if the metadata header should be written.</param>
+        /// <param name="metadata">Additional metadata that is appended to the standard metadata header. If nometata is true, additional metadata is not written.</param>
+        /// <returns>FileInfo object of the created file</returns>
+        internal FileInfo Write(CommandHelp help, bool noMetadata, Hashtable? metadata = null)
+        {
+            try
+            {
+                if (!noMetadata)
+                {
+                    WriteMetadataHeader(help, metadata);
+                    sb.AppendLine();
+                }
+
+                WriteTitle(help);
+                sb.AppendLine();
+
+                WriteSynopsis(help);
+                sb.AppendLine();
+
+                // this adds an empty line after all parameters. So no AppendLine needed.
+                WriteSyntax(help);
+
+                WriteDescription(help);
+                sb.AppendLine();
+
+                WriteExamples(help);
+                sb.AppendLine();
+
+                WriteParameters(help);
+
+                if (help.Inputs != null)
+                {
+                    WriteInputsOutputs(help.Inputs, Constants.InputsMdHeader);
+                }
+
+                if (help.Outputs != null)
+                {
+                    WriteInputsOutputs(help.Outputs, Constants.OutputsMdHeader);
+                }
+
+                WriteNotes(help);
+
+                WriteRelatedLinks(help);
+
+                using StreamWriter mdFileWriter = new(_filePath, append: false, _encoding);
+
+                mdFileWriter.Write(sb.ToString());
+
+                return new FileInfo(_filePath);
+            }
+            finally
+            {
+                Constants.StringBuilderPool.Return(sb);
```

Comment:
By using `IDisposal`, you can write the following code
```
using var writer = new CommandHelpMarkdownWriter(...);
writer.Write(...)
```

Reply:
Fixed